Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -2386,6 +2386,19 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
             if (isInt32ConvertedToDouble(node->child1()) && isInt32ConvertedToDouble(node->child2()))
                 m_state.setShouldTryConstantFolding(true);
         }
+
+        if (node->isBinaryUseKind(Int52RepUse)) {
+            auto isInt32ConvertedToInt52 = [&](Edge& edge) {
+                if (edge->op() == Int52Constant)
+                    return edge->constant()->value().isInt32AsAnyInt();
+                if (edge->op() == Int52Rep)
+                    return edge->child1().useKind() == Int32Use;
+                return false;
+            };
+
+            if (isInt32ConvertedToInt52(node->child1()) && isInt32ConvertedToInt52(node->child2()))
+                m_state.setShouldTryConstantFolding(true);
+        }
```

Comment:
We will remove this part once https://github.com/WebKit/WebKit/pull/42537 gets landed.